### PR TITLE
skimpdf: init at 1.7.2

### PR DESCRIPTION
**Pull request by YvesStraten**
## Description of changes
Init skimpdf in nixpkgs

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sk/skimpdf/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    cp -R *.app $out/Applications
+  '';
+
+  meta = with lib; {
+    description = "Skim is a PDF reader and note-taker for OS X";
+    homepage = "https://skim-app.sourceforge.io/";
+    license = licenses.bsd0;
+    mainProgram = "Skim.app";
+    maintainers = with maintainers; [ YvesStraten ];
+    platforms = [ "x86_64-darwin" "aarch64-darwin" ];
```
Comment (quantenzitrone) [inline pkgs/by-name/sk/skimpdf/package.nix:32]:
```suggestion
    platforms = platforms.darwin;
```

This corresponds to `[ "x86_64-darwin" "i686-darwin" "aarch64-darwin" "armv7a-darwin" ]`.

**Review by quantenzitrone — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sk/skimpdf/package.nix (hunk 1 above)
Comment (YvesStraten) [reply pkgs/by-name/sk/skimpdf/package.nix:32]: Understood, adding changes

File: pkgs/by-name/sk/skimpdf/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    sha256 = "0SeS41ExCZfGmjiK2KvvdFpP7IVzm7xHKCaKmEUYQAY=";
+  };
+
+  nativeBuildInputs = [ undmg ];
+
+  sourceRoot = ".";
+
+  installPhase = ''
+    mkdir -p $out/Applications
+    cp -R *.app $out/Applications
```
Comment (raspher) [inline pkgs/by-name/sk/skimpdf/package.nix:23]:
```suggestion
    runHook preInstall
    mkdir -p $out/Applications
    cp -R *.app $out/Applications
    runHook postInstall
```
https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases

File: pkgs/by-name/sk/skimpdf/package.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Skim is a PDF reader and note-taker for OS X";
+    homepage = "https://skim-app.sourceforge.io/";
+    license = licenses.bsd0;
+    mainProgram = "Skim.app";
+    maintainers = with maintainers; [ YvesStraten ];
+    platforms = platforms.darwin;
```
Comment (raspher) [inline pkgs/by-name/sk/skimpdf/package.nix:32]: afaik this is not builded from source, so please add `sourceProvenance`
https://nixos.org/manual/nixpkgs/stable/#sec-meta-sourceProvenance

File: pkgs/by-name/sk/skimpdf/package.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  installPhase = ''
+    mkdir -p $out/Applications
+    cp -R *.app $out/Applications
+  '';
+
+  meta = with lib; {
+    description = "Skim is a PDF reader and note-taker for OS X";
+    homepage = "https://skim-app.sourceforge.io/";
+    license = licenses.bsd0;
+    mainProgram = "Skim.app";
```
Comment (raspher) [inline pkgs/by-name/sk/skimpdf/package.nix:30]: If there is one `mainProgram` you shouldn't copy multiple binaries to $out (`cp -R *.app $out/Applications` in installPhase)

If there are many executables, this `meta.mainProgram` should not be defined

https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

**Review by raspher — CHANGES_REQUESTED** [review]

**Comment by YvesStraten** [comment]
Hello, @raspher, thank you very much for your suggestions. I have added them to the derivations and squashed the commits accordingly. This should hopefully be everything for this package. Thank you very much and have a very nice day/night 🙏

File: pkgs/by-name/sk/skimpdf/package.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, lib
+, undmg
+, fetchurl
+,
```
Comment (raspher) [inline pkgs/by-name/sk/skimpdf/package.nix:5]:
```suggestion
```

File: pkgs/by-name/sk/skimpdf/package.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}:
+stdenv.mkDerivation rec {
+  pname = "Skim";
+  version = "1.6.21";
```
Comment (raspher) [inline pkgs/by-name/sk/skimpdf/package.nix:9]: new version released https://sourceforge.net/projects/skim-app/files/Skim/

Comment (YvesStraten) [reply pkgs/by-name/sk/skimpdf/package.nix:9]: The new version will be added in a new pr, like nixpkgs wants it to be

**Comment by YvesStraten** [comment]
All suggestions have been added. Thank you very much for your suggestions

File: pkgs/by-name/sk/skimpdf/package.nix (hunk 4 above)
Comment (raspher) [reply pkgs/by-name/sk/skimpdf/package.nix:32]: Still relevant

Comment (YvesStraten) [reply pkgs/by-name/sk/skimpdf/package.nix:32]: There is only Skim.app so the mainProgram tag should be fine

Comment (YvesStraten) [reply pkgs/by-name/sk/skimpdf/package.nix:32]: Oh wait sorry, for some reason my changes to the installPhase did not get synced, fixing it right now

Comment (YvesStraten) [reply pkgs/by-name/sk/skimpdf/package.nix:32]: Has been done

**Comment by YvesStraten** [comment]
Everything should be correct and this can be hopefully merged

**Review by RossComputerGuy — APPROVED** [review]
I approb

**Review by tomberek — APPROVED** [review]
Thanks for the contribution!

**Comment by YvesStraten** [comment]
No problem, thank you so much for approving and merging 🙏

**Pull request merged**